src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -49,6 +60,57 @@ internal bool AutomaticRedirection
                 _automaticRedirection = value;
             }
         }
+        internal bool UseProxy
+        {
+            get
+            {
+                if (_proxyPolicy == CurlProxyUsePolicy.UseDefaultProxy)
+                {
+                    return false;
+                }
+                return _proxyPolicy != CurlProxyUsePolicy.DoNotUseProxy;
```

Comment:
Also, your code is invalid.  Your handler SHOULD NOT over-write the `Proxy` property based on the `UseProxy` Boolean.